Automattic/newspack-plugin
# feat(ia): audience checkout/payment and donations wizard

### All Submissions:






### Changes proposed in this Pull Request:

Implements:
 - Audience Setup Checkout & Payment tab
 - Audience Donations wizard

Most of the effort around this PR is decoupling logic from the existing "Reader Revenue" wizard and placing in different sections of the Audience wizard:

 - Salesforce settings have moved to the Audience Setup tab and should now only render if the publisher already have configured keys for it
 - Payment gateway configuration and checkout billing fields have moved to the Audience Checkout & Payment tab
 - Transaction fees have moved to Audience Donation
 - The Donate block has been updated to use the new API endpoints (see https://github.com/Automattic/newspack-blocks/pull/1974)

<img width="1328" alt="image" src="https://github.com/user-attachments/assets/0c6ac444-d902-42b8-ab6d-94314e1a9ce5">

<img width="1331" alt="image" src="https://github.com/user-attachments/assets/19762eed-20a7-4246-b42f-ffb85b2a9f56">

<img width="1331" alt="image" src="https://github.com/user-attachments/assets/76ba6e9e-bb89-41f9-9cc5-bf3e79bf5b65">

### How to test the changes in this Pull Request:

1. Reset your RAS setup (CLI: `wp option delete newspack_reader_activation_enabled`)
2. Navigate to "Audience -> Setup" and confirm the wizard renders as in the image above
3. Go through each setup step and confirm they work as expected
4. On the "Reader Revenue" step, confirm that clicking the "Update Reader Revenue settings" button redirects to the Checkout & Payment view with a notification to return to the RAS setup:

<img width="1156" alt="image" src="https://github.com/user-attachments/assets/be10e185-232e-4d4b-89d9-c7dc57d32e0b">

5. Complete the RAS setup
6. In "Audience -> Setup", scroll down and confirm the Salesforce settings do not render (if you don't have Salesforce settings already)
7. Add a Salesforce client ID to the database (CLI: `wp option update newspack_salesforce_client_id 123`), refresh and confirm the Salesforce settings render
8. Navigate to the Checkout & Payment tab, switch platforms and confirm it behaves as expected
9. For each platform, tweak the settings, save and refresh the page to confirm the options persist
10. Navigate to Audience -> Donations and confirm the wizard renders as expected
11. Tweak the donation options, save and confirm it behaves as expected
12. Click to edit the donation page (make sure you have https://github.com/Automattic/newspack-blocks/pull/1974)
13. Confirm the Donate block render correctly in the editor and you are able to tweak its options
14. Back to the Audience -> Donations wizard, click the "Revenue" tab and confirm the link redirects to WC's reports admin page

### Other information:
